sycl/test/devicelib/assert.cpp

```diff
@@ -160,9 +171,19 @@ int main() {
     if (WIFEXITED(status) && WEXITSTATUS(status) == EXIT_SKIP_TEST) {
       return 0;
     }
-    if (!WIFSIGNALED(status)) {
-      fprintf(stderr, "error: process did not terminate by a signal\n");
-      return 1;
+    if (getenv("SHOULD_CRASH")) {
+      if (!WIFSIGNALED(status)) {
+        fprintf(stderr, "error: process did not terminate by a signal\n");
+        return 1;
+      }
+    } else {
+      if (WIFSIGNALED(status)) {
+        fprintf(stderr, "error: process should not terminate\n");
```

Comment:
Can't this message interleave with other output?

Reply:
If devicelib assert prints to stderr - yes, it can. This should not be a problem though, because if we hit this, the test is going to fail anyway.